src/main/java/thePackmaster/ui/EnhanceBonfireOption.java

```diff
@@ -0,0 +1,94 @@
+package thePackmaster.ui;
+
+import basemod.ReflectionHacks;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.helpers.ImageMaster;
+import com.megacrit.cardcrawl.helpers.MathHelper;
+import com.megacrit.cardcrawl.localization.UIStrings;
+import com.megacrit.cardcrawl.ui.campfire.AbstractCampfireOption;
+import thePackmaster.SpireAnniversary5Mod;
+import thePackmaster.packs.GemsPack;
+import thePackmaster.vfx.gemspack.SocketGemEffect;
+
+
+public class EnhanceBonfireOption extends AbstractCampfireOption {
+    public static final String[] DESCRIPTIONS;
+    private static final UIStrings UI_STRINGS;
+
+    static {
+        UI_STRINGS = CardCrawlGame.languagePack.getUIString(SpireAnniversary5Mod.makeID("EnhanceBonfireOptions"));
+        DESCRIPTIONS = UI_STRINGS.TEXT;
+
+    }
+
+    //private ArrayList<String> idleMessages;
+    public EnhanceBonfireOption(boolean active) {
+        this.label = DESCRIPTIONS[0];
+
+        for (AbstractCard c : AbstractDungeon.player.masterDeck.group) {
+            c.update();
+        }
+
+        this.usable = active;
+        if (active) {
+            this.description = DESCRIPTIONS[1];
+            this.img = ImageMaster.loadImage(SpireAnniversary5Mod.makeImagePath("ui/scrapcampfire.png"));
+        } else {
+            this.img = ImageMaster.loadImage(SpireAnniversary5Mod.makeImagePath("ui/scrapcampfiredisabled.png"));
+            if (AbstractDungeon.player.gold < 10){
```

Comment:
Is this the socketing gold requirement? You have a static variable that defines it in your pack iirc, Why not use that? This seems out of date